### ISSUE TITLE
[BUG] Using MinMaxMeanDose constraint on structure with only 1 voxel failes

### ISSUE THREAD

**Issue by zperko** [e0404/matRad#601]
**Describe the bug**
When applying a MinMaxMean dose constraint on a structure that only contains a single voxel the code fails when trying to calculate the Jacobian. 

**To Reproduce**
1. Simply apply a constraint on a structure with only one voxel.

** To solve: in matRad_getJacobianStructure.m, line 51 should be 
jacobStruct = [jacobStruct; repmat(spones(mean(dij.physicalDose{1}(cst{i,4}{1},:),1)),nRows,1)];

mean across the rows must be specified, otherwise if only 1 voxel is present in the constraint the dimensions will be incorrect.

**Expected behavior**
That the constraint works regardless of the number of voxels in the structure.

**Desktop (please complete the following information):**
 - OS: Windows
 - Environment & Version Matlab 2021b
 - matRad Version 2.10.1

**Additional context**
Could also be the result of overlap priorities, if after removing higher priority voxels only one remains.

**Comment by wahln** [comment]
Good catch @zperko , thanks for the report and suggested fix (and for following the template). 
Reproduced it on the TG119 phantom, fix works as expected.
Found another bug that occurs when a constraint is applied to an empty structure (which can happen due to doseGrid resizing) caused in https://github.com/e0404/matRad/blob/cb6b04a603533a5906617d489eb36e4a054eaa3a/optimization/%40matRad_OptimizationProblem/matRad_getConstraintBounds.m#L42 - should be `~any(cellfun(@isempty,cst{j,4}))`. In scenario based optimization with multiple ct scenarios, we may need to treat this more carefully - in any case, I will include a warning for that and create a fix.

**Comment by zperko** [comment]
Yeah also had that issue. First work-around is to check the cst structure before starting the optimization, and if any is empty but has a constraint/objective, just delete the constraint/objective. Problem is that emptying of a structure can also happen in setting the priorities, if a higher priority structure completely encompasses a lower priority object, so that needs better handling indeed.

**Comment by wahln** [comment]
yes, I think this use case should now be covered as well. It never happened because with objectives in the scalarized weighted sum approach, this error doesn't occur, and we never happened to assign a constraint to a likely overlapped structure or similar. Reminder to myself - need to write some unit tests to confirm this behavior sustainably.
For the fix, the constraint will not "act" in such a situation of 0 or 1 voxel, but given how the overlap is intended to work, it's probably OK to just warn for this but not interfere otherwise.
